.pre-commit-config.yaml

```diff
@@ -0,0 +1,20 @@
+repos:
+- repo: https://github.com/pre-commit/pre-commit-hooks
+  rev: v4.5.0
```

Comment:
so, the thing is: pre-commit works exclusively locally (except if hooked up in the CI). It keeps a cache of the packages and if you just track main that means it will updated the cache on every `git commit` [^1] to check for a new version. version pinning is almost required here but since it's a local "check before you push" IMO it doesn't matter too much - and that's why the CI runs doing ruff/pylint are still important.

[^1]: ok, that's what it did a few months ago when I discovered it

Reply:
`pre-commit autoupdate` updates the pins to the latest version (see [the docs](https://pre-commit.com/#pre-commit-autoupdate)), so I guess we could run this as CI job? Or just have a calendar reminder every 6m to do this :)

Reply:
Well, without server-side CI/dependabot etc. the only plausible way for this is that the pre-commit CLI itself warns about outdated versions, and suggests `auto-update`. If that happens, fine for me.

Reply:
I think pre-commit stems from a programming culture where pinning versions is common, so I don't think it has that particular option, or at least I haven't found it yet